flink-connectors/flink-connector-aws-kinesis-data-streams/src/main/java/org/apache/flink/connector/kinesis/sink/KinesisDataStreamsSinkWriter.java

```diff
@@ -56,6 +62,61 @@
 class KinesisDataStreamsSinkWriter<InputT> extends AsyncSinkWriter<InputT, PutRecordsRequestEntry> {
     private static final Logger LOG = LoggerFactory.getLogger(KinesisDataStreamsSinkWriter.class);
 
+    private static final RetryValidationStrategy INVALID_CREDENTIALS_STRATEGY =
+            new RetryValidationStrategy(
+                    err -> ExceptionUtils.findThrowable(err, StsException.class).isPresent(),
```

Comment:
nit: This is much better, but still duplicating logic. You could sub `RetryValidationStrategy` with a `RootCauseRetryValidationStrategy` that wraps up the `ExceptionUtils.findThrowable(` and you simply pass the class to look for